### PR TITLE
Time to deal with these deprecations

### PR DESCRIPTION
**Pull request by dansanduleac**
## Before this PR

Ignoring deprecations causing other projects to have to ignore them too when testing against sls-packaging on gradle 6.

## After this PR
==COMMIT_MSG==
Stop ignoring deprecations so we know we're clean on gradle 6.

Also reverts #508 as I haven't seen the reported failure anymore.
==COMMIT_MSG==

### CODE REVIEW

**Comment by dansanduleac** [comment]
@ferozco re #508, I've run this in SSH and can't see any defunct processes after the tests have finished running:

```
$ ps auxwf
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
circleci     7  0.6  0.0 127300 31100 ?        Ssl  23:25   0:03 /bin/circleci-agent --config /.circleci-runner-config.json --task-data /.circleci-task-data --outerServerUrl https://circleci-internal-outer-build-agent:5500 _internal runner --sshAdvertiseAddr=54.167.234.40:64536 --sshBindPort=54782
circleci  2606  0.0  0.0  20000  3588 pts/1    Ss   23:29   0:00  \_ -bash
circleci  5272  0.0  0.0  38384  3176 pts/1    R+   23:35   0:00  |   \_ ps auxwf
circleci  4633  0.0  0.0  20000  3804 pts/2    Ss+  23:34   0:00  \_ -bash
circleci     1  0.0  0.0   1052     4 pts/0    Ss   23:25   0:00 /dev/init -- /bin/sh
circleci     6  0.0  0.0   4280   676 pts/0    S+   23:25   0:00 /bin/sh
circleci    99  7.0  0.9 6237384 660892 ?      Ssl  23:25   0:42 /usr/local/openjdk-8/bin/java -XX:MaxMetaspaceSize=256m -Dfile.encoding=UTF-8 -Duser.country -Duser.language=en -Duser.variant -cp /home/circleci/.gradle/wrapper/dists/gradle-6.4-20200318141820+0000-bin/4c6zdk8w06p9yej6yian3khjw/gradle-6.4-20200318141820+0000/lib/gradle-launcher-6.4.jar org.gradl
```

File: gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.java [hunk 1]
```diff
@@ -260,12 +260,61 @@ public void execute(Task _task) {
             }
         }));
 
-        project.afterEvaluate(p -> DistTarTask.configure(
-                distTar.get(),
-                project,
-                distributionExtension.getDistributionServiceName().get(),
-                distributionExtension.getExcludeFromVar().get(),
-                distributionExtension.getEnableManifestClasspath().get()));
+        project.afterEvaluate(proj -> distTar.configure(task -> {
```
Comment (ferozco) [inline gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.java:262]: awesome to get rid of a bit more groovy, but could we keep this separate just to keep the plugin clean

Comment (dansanduleac) [reply gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.java:262]: moved

**Comment by ferozco** [comment]
👍

**Pull request merged**